### PR TITLE
planner: fix bug that window function do not check ignore nulls.

### LINKED ISSUE #10556: `IGNORE NULLS` in window function not implemented but no error thrown
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
SELECT FIRST_VALUE(i) IGNORE NULLS OVER () FROM t1;
```

2. What did you expect to see?

Error thrown, `This version of TiDB hasn't supported IGNORE NULLS currently`.

3. What did you see instead?

executes successfully.

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve? 
fix #10556 

### What is changed and how it works?
Add a simple judgment.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - None

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

File: planner/core/logical_plan_builder.go [hunk 1]
```diff
@@ -3329,6 +3329,9 @@ func (b *PlanBuilder) buildWindowFunctions(ctx context.Context, p LogicalPlan, g
 // Because of the grouped specification is different from it, we should especially check them before build window frame.
 func (b *PlanBuilder) checkOriginWindowSpecs(funcs []*ast.WindowFuncExpr, orderByItems []property.Item) error {
 	for _, f := range funcs {
+		if f.IgnoreNull {
```
Comment (lamxTyler) [inline planner/core/logical_plan_builder.go]: The `Distinct` and `FromLast` should also be checked, you can do it in this pr or another pr.

Comment (lzmhhh123) [reply planner/core/logical_plan_builder.go]: OK, I'll do it in this pr.

**Pull request merged**